compiler/rustc_passes/src/dead.rs

```diff
@@ -251,7 +253,10 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
         dotdot: hir::DotDotPos,
     ) {
         let variant = match self.typeck_results().node_type(lhs.hir_id).kind() {
-            ty::Adt(adt, _) => adt.variant_of_res(res),
+            ty::Adt(adt, _) => {
+                self.check_def_id(adt.did());
```

Comment:
Could you add a comment?

Reply:
Could you change the comment to explain *why* we mark the adt live in that case?